tests/libcxx/expected_results.txt

```diff
@@ -567,10 +567,6 @@ std/re/re.traits/transform.pass.cpp FAIL
 # STL bug: Incorrect return types.
 std/numerics/complex.number/cmplx.over/pow.pass.cpp FAIL
 
-# STL bug: Missing <valarray> assignment operators.

```

Comment:
Out of curriosity there are no tests for `indirect_array` or `gslice_array`?

Reply:
Good question.
[The test for indirect_array](https://github.com/llvm/llvm-project/blob/d66428cb995c7a04ecb02951d5021d815fc02b2b/libcxx/test/std/numerics/numarray/template.slice.array/slice.arr.assign/slice_array.pass.cpp#L23-L35)

```
    int a1[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
    int a2[] = {-1, -2, -3, -4, -5, -6, -7, -8, -9, -10, -11, -12};
    std::valarray<int> v1(a1, sizeof(a1)/sizeof(a1[0]));
    const std::valarray<int> v2(a2, sizeof(a2)/sizeof(a2[0]));
    v1[std::slice(1, 5, 3)] = v2[std::slice(2, 5, 2)];
    assert(v1.size() == 16);
    assert(v1[ 0] ==  0);
    assert(v1[ 1] == -3);
    assert(v1[ 2] ==  2);
    assert(v1[ 3] ==  3);
    assert(v1[ 4] == -5);
    assert(v1[ 5] ==  5);
    assert(v1[ 6] ==  6);
```
is similar to the [test of slice_array](https://github.com/llvm/llvm-project/blob/d66428cb995c7a04ecb02951d5021d815fc02b2b/libcxx/test/std/numerics/numarray/template.indirect.array/indirect.array.assign/indirect_array.pass.cpp#L22-L43):
```
    int a1[] = { 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11,
                12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23,
                24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35,
                36, 37, 38, 39, 40};
    const std::size_t N1 = sizeof(a1)/sizeof(a1[0]);
    int a2[] = { -0,  -1,  -2,  -3,  -4,  -5,  -6,  -7,  -8,  -9, -10, -11,
                -12, -13, -14, -15, -16, -17, -18, -19, -20, -21, -22, -23,
                -24, -25, -26, -27, -28, -29, -30, -31, -32, -33, -34, -35,
                -36, -37};
    const std::size_t N2 = sizeof(a2)/sizeof(a2[0]);
    std::size_t s[] = { 3,  4,  5,  7,  8,  9, 11, 12, 13, 15, 16, 17,
                       22, 23, 24, 26, 27, 28, 30, 31, 32, 34, 35, 36};
    const std::size_t S = sizeof(s)/sizeof(s[0]);
    std::valarray<int> v1(a1, N1);
    const std::valarray<int> v2(a2, N2);
    std::valarray<std::size_t> ia(s, S);
    v1[ia] = v2[ia];
    assert(v1.size() == 41);
    assert(v1[ 0] ==  0);
    assert(v1[ 1] ==  1);
    assert(v1[ 2] ==  2);
    assert(v1[ 3] == -3);
```

The difference is that [slice_array also tests assignment](https://github.com/llvm/llvm-project/blob/d66428cb995c7a04ecb02951d5021d815fc02b2b/libcxx/test/std/numerics/numarray/template.slice.array/slice.arr.assign/slice_array.pass.cpp#L46-L55) :
```
    // Test return value of assignment.
    {
    int a1[] = {0, 1, 2};
    std::valarray<int> v1(a1, 3);
    std::slice_array<int> s1 = v1[std::slice(1, 1, 1)];
    std::slice_array<int> s2 = v1[std::slice(0, 1, 1)];
    std::slice_array<int> const & s3 = (s1 = s2);
    assert(&s1 == &s3);
    }
```

The common part of `slice_array` and `indirect_array` test passed before this PR, since it actually tests what I call in https://github.com/microsoft/STL/pull/988#issuecomment-654020141 `v = a`, but not `v = v`.

The test uses `v = a`, and not `v = v` because of `const` version being selected:
https://github.com/microsoft/STL/blob/5be7d49c243979231dff35919d3a3d813d75d714/stl/inc/valarray#L309-L311
I'm not sure if they meant to test exactly `const` version, or it just happened this way.